lightning/src/ln/blinded_payment_tests.rs

```diff
@@ -2298,9 +2298,230 @@ fn test_trampoline_single_hop_receive() {
 	do_test_trampoline_single_hop_receive(false);
 }
 
-fn do_test_trampoline_unblinded_receive(success: bool) {
-	// Simulate a payment of A (0) -> B (1) -> C(Trampoline) (2)
+#[derive(Copy, Clone, PartialEq, Eq)]
+enum TrampolineTestCase {
+	Success,
+	Underpayment,
+	OuterCLTVLessThanTrampoline,
+}
+
+impl<'a> TrampolineTestCase {
+	fn payment_failed_conditions(
+		self, final_payment_amt: &'a [u8], final_cltv_delta: &'a [u8],
+	) -> Option<PaymentFailedConditions<'a>> {
+		match self {
+			TrampolineTestCase::Success => None,
+			TrampolineTestCase::Underpayment => {
+				Some(PaymentFailedConditions::new().expected_htlc_error_data(
+					LocalHTLCFailureReason::FinalIncorrectHTLCAmount,
+					final_payment_amt,
+				))
+			},
+			TrampolineTestCase::OuterCLTVLessThanTrampoline => {
+				Some(PaymentFailedConditions::new().expected_htlc_error_data(
+					LocalHTLCFailureReason::FinalIncorrectCLTVExpiry,
+					final_cltv_delta,
+				))
+			},
+		}
+	}
+
+	fn expected_log(&self) -> Option<(&str, &str, usize)> {
```

Comment:
I would slightly prefer to in-line this method

Reply:
Happy to inline - just curious why only this one?

There's an argument to inline a few of these methods (since they're quite thin), but I find it helpful for readability to have all the test case specific logic in one place.

Reply:
That's fair, this was one method I kept jumping back and forth to read so it stood out. But it is inconsistent to inline just this one so nbd.